src/vm/threads.cpp

```diff
@@ -2144,6 +2144,30 @@ HANDLE Thread::CreateUtilityThread(Thread::StackSizeBucket stackSizeBucket, LPTH
 }
 
 
+// Represent the value of OVERRIDE_DEFAULT_STACK_SIZE_4KPAGES as passed in the property bag to the host during construction
+static SIZE_T s_overrideDefaultStackSize = 0;
+
+void OverrideDefaultStackSize(LPCWSTR valueStr)
+{
+    if (valueStr)
+    {
+        LPWSTR end;
+        errno = 0;
+        unsigned long value = wcstoul(valueStr, &end, 10);
+
+        SIZE_T max4KPages = SIZE_T(1) << (31 - 12);
+
+        if ((errno == ERANGE) || (valueStr == end) || (end == nullptr) || (end[0] != 0) || (value >= max4KPages))
+        {
+            ThrowMessage("OVERRIDE_DEFAULT_STACK_SIZE_4KPAGES value parse error");
```

Comment:
I don't know much about exceptions in the runtime... so maybe this is correct, but I find it weird that `ThrowMessage` is not used anywhere else as far as I can tell. And also then there's this comment:
https://github.com/dotnet/coreclr/blob/e0fd735a866861b35679ac503193809a305fac2b/src/vm/exceptmacros.h#L481

Another consideration is the error message, I would think it should be localized (as in coming from resources somewhere?).

Reply:
```
#define ThrowMessage        "Don't use this in the VM directory" 
```
That is a terrible worthless define.  It should be a static assert at least. Silently ignoring the error is just silly.  More technical debt...

I didn't expect this to be a user facing error. I think the message will get lost anyway.  It will likely become one of those E_FAIL startup errors that are so nice to debug....

Looks like this might have to become a `COMPlusThrowHR`...